docker/Dockerfile

```
@@ -1,21 +1,22 @@
-# BUILDER
 FROM ubuntu:22.04
-WORKDIR /builder
+
+ARG GPU_CHOICE=${GPU_CHOICE:-A} # A: NVIDIA, B: AMD, C: APPLE, D: INTEL, N: NONE (CPU)
 ARG TORCH_CUDA_ARCH_LIST="${TORCH_CUDA_ARCH_LIST:-3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX}"
 ARG BUILD_EXTENSIONS="${BUILD_EXTENSIONS:-}"
 ARG APP_UID="${APP_UID:-6972}"
 ARG APP_GID="${APP_GID:-6972}"
 
+WORKDIR /builder
 RUN --mount=type=cache,target=/var/cache/apt,sharing=locked,rw \
     apt update && \
     apt install --no-install-recommends -y git vim build-essential python3-dev pip bash curl && \
     rm -rf /var/lib/apt/lists/*
 WORKDIR /home/app/
-RUN git clone https://github.com/oobabooga/text-generation-webui.git 
+RUN git clone https://github.com/oobabooga/text-generation-webui.git
 WORKDIR /home/app/text-generation-webui
-RUN GPU_CHOICE=B USE_CUDA118=FALSE LAUNCH_AFTER_INSTALL=FALSE INSTALL_EXTENSIONS=TRUE ./start_linux.sh --verbose
+RUN USE_CUDA118=FALSE LAUNCH_AFTER_INSTALL=FALSE INSTALL_EXTENSIONS=TRUE ./start_linux.sh --verbose
 COPY CMD_FLAGS.txt /home/app/text-generation-webui/
 EXPOSE ${CONTAINER_PORT:-7860} ${CONTAINER_API_PORT:-5000} ${CONTAINER_API_STREAM_PORT:-5005}
-WORKDIR /home/app/text-generation-webui
+ENV HOME=/home/app/text-generation-webui
 # set umask to ensure group read / write at runtime
-CMD umask 0002 && export HOME=/home/app/text-generation-webui && ./start_linux.sh
+CMD umask 0002 && ./start_linux.sh $CLI_ARGS
```

- `umask 0002` seems unnecessary? Why is group write permission needed at runtime? [it was introduced 9 months ago here, but no insights as to why](https://github.com/oobabooga/text-generation-webui/pull/4768/files#diff-52c6e96914b8b24ca636ecf09a028a313c8790830d80d40969fcaf911cacb140R44-R56).
- [`CLI_ARGS` was removed in March](https://github.com/oobabooga/text-generation-webui/issues/5422). Regardless, this is not the correct approach with Docker.

```suggestion
ENTRYPOINT [ "/home/app/text-generation-webui/start_linux.sh" ]
```

Now `CLI_ARGS` would just be provided as the command (`CMD`):

```
docker run --rm -it image-name $CLI_ARGS
```

In `compose.yaml` the `command` field can set the CLI args to use, or you can use the Docker Compose ENV interpolation from a `.env` file to also use `${CLI_ARGS}` as a value (kinda redundant though).